src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.factory.BetDaoFactory;
+import core.basesyntax.factory.UserDaoFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Class<BetDaoImpl> betDaoImplClass = BetDaoImpl.class;
+        Class<UserDaoImpl> userDaoImplClass = UserDaoImpl.class;
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                if (betDaoImplClass.getAnnotation(Dao.class) != null
```

Comment:
```suggestion
                if (betDaoImplClass.isAnnotationPresent(Dao.class)
```
можна ще так перевіряти